src/ui/profile/ProfileSelection.js

```diff
@@ -0,0 +1,170 @@
+// eslint-disable-next-line no-redeclare
+/* global chrome */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/profile.module.css';
+import AddProfile from './AddProfile';
+import BeyondLoader from './BeyondLoader';
+import Profiles from './Profile';
+
+const LoadedProfile = (profileModel) => ({
+  status: 'loaded',
+  profile: profileModel,
+});
+
+const ErrorProfile = (error, profile) => ({
+  status: 'error',
+  error,
+  profile,
+});
+
+const LoadingProfile = (profile) => ({
+  status: 'loading',
+  profile,
+});
+
+function ProfileSelection(props) {
```

Comment:
Another place you can destructure props in argument position:

```js
function ProfileSelection({ onCharacterReady}) { ...
```

This is a pretty standard convention for function components.